operations/services.md

```diff
@@ -1579,6 +1579,311 @@ Returns all companionships based on customers, reservations or reservation group
+| `Room` | bool | required | Specify whether [Resource access token](#resource-access-token) grants permission to access room. |
+| `Floor` | bool | required | Specify whether [Resource access token](#resource-access-token) grants permission to access floor. |
+| `Building` | bool | required | Specify whether [Resource access token](#resource-access-token) grants permission to access building. |
+
+## Add resource access tokens
+
+Adds new resource access tokens with the specified data.
+
+### Request
+
+`[PlatformAddress]/api/connector/v1/resourceAccessTokens/add`
+
+```javascript
+{
+    "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
+    "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
+    "Client": "Sample Client 1.0.0",
+    "ResourceAccessTokenParameters": [
+        {
+            "ServiceOrderId": "35d4b117-4e60-44a3-9580-c582117eff98",
+            "CompanionshipId": "25d4b117-4e60-44a3-9580-c582117eff98",
+            "ResourceId": "65d4b117-4e60-44a3-9580-c582117eff98",
+            "Type": "PinCode",
+            "Value": "1234#",
+            "SerialNumber": null,
+            "ValidityStartUtc": "2020-10-09T22:00:00Z",
+            "ValidityEndUtc": "2020-10-10T22:00:00Z",
+            "Permissions":
+            {
+                "Bed": { "Value": true },
+                "Room": { "Value": false },
+                "Floor": { "Value": false },
+                "Building": { "Value": false }
+            }
+        }
+    ]
+}
+```
+
+| Property | Type | Contract | Description |
+| --- | --- | --- | --- |
+| `ClientToken` | string | required | Token identifying the client application. |
+| `AccessToken` | string | required | Access token of the client application. |
+| `Client` | string | required | Name and version of the client application. |
+| `ResourceAccessTokenParameters` | Array of [Resource access token parameter](#resource-access-token-parameter) | required, max 1000 items | Parameters of [Resource access token](#resource-access-token). |
+
+#### Resource access token parameter
+
+| Property | Type | Contract | Description |
+| --- | --- | --- | --- |
+| `ServiceOrderId` | string | required | Unique identifier of a service order (for example [Reservation](reservations.md#reservation)). |
+| `CompanionshipId` | string | optional | Unique identifier of [Companionship](services.md#companionship). |
+| `ResourceId` | string | optional | Unique identifier of [Resource](enterprises.md#resource). |
+| `Type` | [Resource access token type](#resource-access-token-type) | required | Type of stored value. |
+| `Value` | string | required | Value of resource access token |
+| `SerialNumber` | string | optional | Serial number of [Resource access token type](#resource-access-token-type). |
+| `ValidityStartUtc` | string | required | Marks the start of interval in which the resource access token can be used. |
```

Comment:
Inconsistent documentation of datetime, we usually specify format.

Reply:
BTW instead of "Marks the start" I'd go with simpler "Start of the interval in which the..."